src/main/scala/mesosphere/marathon/state/Instance.scala

```diff
@@ -0,0 +1,80 @@
+package mesosphere.marathon
+package state
+
+import core.instance.{Reservation, Instance => CoreInstance}
+import mesosphere.marathon.core.condition.Condition
+import mesosphere.marathon.core.instance.Instance.InstanceState
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.raml.Raml
+import play.api.libs.functional.syntax._
+import play.api.libs.json.Reads._
+import play.api.libs.json._
+
+case class Instance(
+    instanceId: CoreInstance.Id,
+    agentInfo: Option[CoreInstance.AgentInfo],
+    state: InstanceState,
+    tasksMap: Map[Task.Id, Task],
+    runSpecVersion: Timestamp,
+    unreachableStrategy: UnreachableStrategy,
+    reservation: Option[Reservation]) extends MarathonState[Protos.Json, Instance] {
+
+  val isReserved: Boolean = state.condition == Condition.Reserved
+
+  def isReservedTerminal: Boolean = isReserved
+
+  override def mergeFromProto(message: Protos.Json): Instance = {
+    Json.parse(message.getJson).as[Instance]
+  }
+  override def mergeFromProto(bytes: Array[Byte]): Instance = {
+    mergeFromProto(Protos.Json.parseFrom(bytes))
+  }
+  override def toProto: Protos.Json = {
+    Protos.Json.newBuilder().setJson(Json.stringify(Json.toJson(this))).build()
+  }
+  override def version: Timestamp = runSpecVersion
+
+  def toCoreInstance(runSpec: RunSpec) = CoreInstance(instanceId, agentInfo, state, tasksMap, runSpec, reservation)
```

Comment:
I thought about `Instance(stateInstance, runSpec)`. However, that leaks the state into our internal logic. I would like to avoid that. Also, our internal instance should not really know about the state instance. It does not care how it is persisted.